doc/source/ray-air/preprocessors.rst

```diff
@@ -0,0 +1,175 @@
+    :end-before: __preprocessor_fit_transform_end__
+
+Finally, call ``transform_batch`` on a single batch of data.
+
+.. literalinclude:: doc_code/preprocessors.py
+    :language: python
+    :start-after: __preprocessor_transform_batch_start__
+    :end-before: __preprocessor_transform_batch_end__
+
+Life of an AIR Preprocessor
+---------------------------
+
+Now that we've gone over the basics, let's dive into how Preprocessors fit into an end-to-end application built with AIR.
+
+.. figure:: images/air-preprocessor.svg
+
+Throughout this section, we'll go through an example using XGBoost. The same logic is applicable to other integrations as well.
+
+Trainer
+~~~~~~~
+
+The journey of the Preprocessor starts with the ``Trainer``. If the Trainer is instantiated with a Preprocessor,
+then the following logic will be executed when ``Trainer.fit()`` is called:
+
+#. If a ``"train"`` Dataset is passed in, then the Preprocessor will call ``fit()`` on it.
+#. The Preprocessor will then call ``transform()`` on *all* Datasets, including the ``"train"`` Dataset.
+#. The Trainer will then perform training on the preprocessed Datasets.
+
+.. literalinclude:: doc_code/preprocessors.py
+    :language: python
+    :start-after: __trainer_start__
+    :end-before: __trainer_end__
+
+.. note::
+
+    If you're passing a Preprocessor that is already fitted, it will be refitted on the ``"train"`` Dataset.
+    Adding the functionality to support passing in a fitted Preprocessor is being tracked
+    `here <https://github.com/ray-project/ray/issues/25299>`__.
+
+Tune
+~~~~
+
+If you're using ``Ray Tune`` for hyperparameter optimization, be aware that each Trial will instantiate its own copy of
+the Preprocessor and the fitting and transformation logic will occur once per Trial.
+
+Checkpoint
+~~~~~~~~~~
+
+``Trainer.fit()`` returns a ``Results`` object which contains a ``Checkpoint``.
+If a Preprocessor was passed into the Trainer, then it will be saved in the Checkpoint along with any fitted state.
+
+As a sanity check, let's confirm the Preprocessor is available in the Checkpoint. In practice you should not need to do this.
+
+.. literalinclude:: doc_code/preprocessors.py
+    :language: python
+    :start-after: __checkpoint_start__
+    :end-before: __checkpoint_end__
+
+
+Predictor
+~~~~~~~~~
+
+A ``Predictor`` can be constructed from a saved ``Checkpoint``. If the Checkpoint contains a Preprocessor,
+then the Preprocessor will be used to call ``transform_batch`` on input batches prior to performing inference.
+
+In the following example, we show the Batch Predictor flow. The same logic applies to the Online Inference flow.
+
+.. literalinclude:: doc_code/preprocessors.py
+    :language: python
+    :start-after: __predictor_start__
+    :end-before: __predictor_end__
+
+Types of Preprocessors
+----------------------
+Ray AIR provides a handful of Preprocessors that you can use out of the box, and more will be added over time.
```

Comment:
To make the subsections more consistent, should we call this tabbed content "Basic Preprocessors" or so, so that on the right it will render as

Basic Preprocessors
Preprocessor Chains (<-- change to this?)
Custom Preprocessors